cprover_bindings/src/goto_program/typ.rs

```diff
@@ -578,76 +614,30 @@ impl Type {
     /// We define it recursively as a type which has exactly one field, which it itself either
     /// a transparent type, or is a scalar type.
     pub fn is_transparent_type(&self, st: &SymbolTable) -> bool {
-        match self {
-            // Follow tags to get the underlying structure
-            StructTag(tag) | UnionTag(tag) => st.lookup(*tag).unwrap().typ.is_transparent_type(st),
-
-            // Recursively check: does this only have one field, which is either a wrapper or base type.
-            Struct { .. } | Union { .. } => self.unwrap_transparent_type(st).is_some(),
-
-            // Base types
-            Array { .. }
-            | Bool
-            | CBitField { .. }
-            | CInteger(_)
-            | Code { .. }
-            | Constructor
-            | Double
-            | Empty
-            | FlexibleArray { .. }
-            | Float
-            | IncompleteStruct { .. }
-            | IncompleteUnion { .. }
-            | InfiniteArray { .. }
-            | Pointer { .. }
-            | Signedbv { .. }
-            | Unsignedbv { .. }
-            | VariadicCode { .. }
-            | Vector { .. } => false,
-        }
+        (self.is_struct_like() || self.is_union_like())
```

Comment:
Why isn't `self.unwrap_transparent_type(st).is_some()` enough?

Reply:
That would match scalar types too.